api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -267,6 +267,10 @@ type HostedControlPlaneStatus struct {
 	// Condition contains details for one aspect of the current state of the HostedControlPlane.
 	// Current condition types are: "Available"
 	// +optional
+	// +listType=map
```

Comment:
Why this change?  not sure if this will cause existing HCs to fail validation on update.

I would like to ensure this doesn't cause an issue on HO/CRD update.

Reply:
This had to change because without it you can't PATCH or SSA one condition, you have to update all of them. I think you guys are doing UPDATE calls now so it won't impact you but at whatever point you go to PATCH or SSA to be more targeted, it will benefit you.